### PR TITLE
Product Page Filters

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added Filters in Product Page,
List of Filters,
1. Search 
2. Sort by Price
3. Sort by Category
4. Sort by Rating
5. Sort by Range
Screenshot (285)

### CODE REVIEW

File: src/Contexts/FilterContext.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      if (sortBy === "LOW_TO_HIGH")
+        return filteredList.sort((a, b) => a.price - b.price);
+
+      if (sortBy === "HIGH_TO_LOW")
+        return filteredList.sort((a, b) => b.price - a.price);
+
+      return filteredList;
+    }
+    return filteredList;
+  };
```
Comment (JantuDeb) [inline src/Contexts/FilterContext.js]: You can extract these function in a separate file

File: src/Contexts/FilterContext.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+  const SearchedList = searchProduct(data, state.searchProdcut);
+  const sortBypriceList = sortbyprice(SearchedList, state.sortbyprice);
+  const sortByCategoryList = sortbycategory(sortBypriceList, state);
+  const sortByRatingList = sortbyrating(sortByCategoryList, state.sortbyrating);
+  const updatedList = sortbyrange(sortByRatingList, state.sortbyrange);
```
Comment (JantuDeb) [inline src/Contexts/FilterContext.js]: You can use a `compose` function here

File: src/Pages/ProductPage/Components/ProductCart.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { v4 as uuidv4 } from "uuid";
+import { useFilter } from "../../../Contexts/FilterContext";
+export const ProductCart = () => {
+  const { updatedList } = useFilter();
+  return (
+    <section className="card-product">
+      {updatedList.map((item) => {
+        return (
+          <div key={uuidv4()} className="card-product__wrapper">
```
Comment (JantuDeb) [inline src/Pages/ProductPage/Components/ProductCart.js:9]: Do not call `uuidv4()` directly on `key`

**Review by JantuDeb — APPROVED** [review]
Suggested some changes

**Review by ShreyashMahajan — APPROVED** [review]
good to merge

**Review by neetukumari4858 — APPROVED** [review]
overall looks good just do some changes which is suggested by @JantuDeb

File: src/Contexts/FilterContext.js (hunk 1 above)
Comment (SaurabhChawda) [reply src/Contexts/FilterContext.js]: Changed in Code ,Thankyou

File: src/Contexts/FilterContext.js (hunk 2 above)
Comment (SaurabhChawda) [reply src/Contexts/FilterContext.js]: I have updated the functions, Thankyou

**Comment by SaurabhChawda** [comment]
> good to merge

Thankyou

**Comment by SaurabhChawda** [comment]
> overall looks good just do some changes which is suggested by @JantuDeb

Okay, Thankyou

**Pull request merged**